.github/workflows/ci.yml

```diff
@@ -0,0 +1,112 @@
+name: Pipeline
+
+on:
+  pull_request:
+    branches:
+      - master
+  push:
+    branches:
+      - test_task
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  ci:
+    runs-on: ubuntu-latest
+    steps:
+      - name: start timer
+        run: |
+         echo "START_TIME=$(date +%s)" >> $GITHUB_ENV
+
+      - name: checkout
+        uses: actions/checkout@v4
+
+      - name: get node.js version from package.json
+        id: get-versions
+        run: echo "::set-output name=node::$(jq -r .version ${{ github.workspace }}/angular/package.json)"
+
+      - uses: actions/setup-node@v1
+        with:
+          node-version: ${{steps.get-versions.outputs.node}}
+
+      - name: install dependencies
+        working-directory: ${{ github.workspace }}/angular
+        run: npm install
```

Comment:
It is better to use `npm ci` instead of `npm install`, cause it won't update packages versions - https://builtin.com/software-engineering-perspectives/npm-ci-vs-npm-install